ext/imap/config.m4

```diff
@@ -233,11 +233,21 @@ if test "$PHP_IMAP" != "no"; then
     dnl utf8_to_mutf7 takes an argument and will segfault without it. We
     dnl therefore test another function utf8_to_mutf7_php() which calls
     dnl the utf8_to_mutf7() function with the empty string as an argument.
-    PHP_IMAP_TEST_BUILD(utf8_to_mutf7_php, [
+    AC_MSG_CHECKING(whether utf8_to_mutf7 function present)
+    old_CPPFLAGS="${CPPFLAGS}"
+    CPPFLAGS="${CPPFLAGS} -I${IMAP_INC_DIR}"
+    PHP_TEST_BUILD(utf8_to_mutf7_php, [
+      AC_MSG_RESULT(yes)
       AC_DEFINE(HAVE_IMAP_MUTF7, 1, [ ])
-    ], [], $TST_LIBS, [
-      char utf8_to_mutf7_php(){ return utf8_to_mutf7(""); }
+    ], [
+      AC_MSG_RESULT(no)
+    ], [
+      $TST_LIBS
+    ], [
+#include <c-client.h>
+char utf8_to_mutf7_php(){ return utf8_to_mutf7(""); }
```

Comment:
Nice trick but now it complains about type-casting because the function return pointer

```
/mnt $ cat /usr/include/imap/utf8aux.h |grep utf8_to_mutf7
unsigned char *utf8_to_mutf7 (unsigned char *src);
```

Reply:
`checking whether utf8_to_mutf7 function present... no`

```
configure:46303: checking whether utf8_to_mutf7 function present
configure:46326: clang -o conftest -O2 -fomit-frame-pointer -g  -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -fvisibility=hidden -Os -fomit-frame-pointer -D_GNU_SOURCE -I/usr/include/imap -Wl,--as-needed,-O1,--sort-common -Wl,-rpath,/lib -L/lib conftest.c 
       -lc-client -lcrypt  -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err -lssl -lcrypto
     -lrt -lm -lacl  -lxml2 -lpcre2-8 -lz >&5
conftest.c:327:48: warning: passing 'char[1]' to parameter of type 'unsigned char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
char utf8_to_mutf7_php(){ return utf8_to_mutf7(""); }
                                               ^~
/usr/include/imap/utf8aux.h:43:46: note: passing argument to parameter 'src' here
unsigned char *utf8_to_mutf7 (unsigned char *src);
                                             ^
conftest.c:327:34: error: incompatible pointer to integer conversion returning 'unsigned char *' from a function with result type 'char' [-Wint-conversion]
char utf8_to_mutf7_php(){ return utf8_to_mutf7(""); }
                                 ^~~~~~~~~~~~~~~~~
1 warning and 1 error generated.
```

Reply:
Argh, and it's using the wrong type because `PHP_TEST_BUILD` declares every function as type `char`. I'll have to use one of the bare autoconf macros, it looks like.

Reply:
For the same reason it fails for `utf8_mime2text`
```
configure:45093: checking for utf8_mime2text signature
configure:45116: clang -c -I/usr/include/imap -Os -fomit-frame-pointer -D_GNU_SOURCE conftest.c >&5
conftest.c:329:32: error: too few arguments to function call, expected 3, have 2
        utf8_mime2text(src, dst);
        ~~~~~~~~~~~~~~         ^
/usr/include/imap/utf8aux.h:37:6: note: 'utf8_mime2text' declared here
long utf8_mime2text (SIZEDTEXT *src,SIZEDTEXT *dst,long flags);
     ^
1 error generated.
```

Reply:
I think the mime2text thing is intended? It's trying to figure out of the signature is the old one with two args or the new one with three.